src/pages/iou/IOUTransactions.js

```diff
@@ -17,33 +18,105 @@ const propTypes = {
-    reportActions,
-    chatReportID,
-    iouReportID,
-}) => (
-    <View style={[styles.mt3]}>
-        {_.map(reportActions, (reportAction) => {
-            if (reportAction.actionName === 'IOU'
-                && reportAction.originalMessage.IOUReportID === iouReportID) {
-                return (
-                    <ReportTransaction
-                        chatReportID={chatReportID}
-                        iouReportID={iouReportID}
-                        action={reportAction}
-                        key={reportAction.sequenceNumber}
-                    />
-                );
+class IOUTransactions extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            rejectableTransactionIDs: [],
+        };
+
+        this.updateRejectableTransactions = this.updateRejectableTransactions.bind(this);
+    }
+
+    componentDidMount() {
+        this.updateRejectableTransactions();
+    }
+
+    componentDidUpdate(oldProps) {
+        if (oldProps.reportActions !== this.props.reportActions) {
+            this.updateRejectableTransactions();
+        }
+    }
+
+    /**
+     * Update the rejectableTransactionIDs state array. A transaction must meet multiple requirements in order to be
+     * rejectable. We must exclude transactions not associated with the iouReportID, actions which have already been
+     * rejected, and those which are not of type 'create').
+     */
+    updateRejectableTransactions() {
+        if (this.props.isReportPaid) {
+            this.setState({rejectableTransactionIDs: []});
+            return;
+        }
+
+        const actionsForIOUReport = _.filter(this.props.reportActions, action => action.originalMessage
+            && action.originalMessage.type && action.originalMessage.IOUReportID === this.props.iouReportID);
+
+        const rejectedTransactions = _.filter(actionsForIOUReport, action => ['cancel', 'decline']
+            .includes(action.originalMessage.type));
+
+        const rejectedTransactionIDs = _.map(rejectedTransactions, (rejectedTransaction) => {
+            const transactionID = lodashGet(rejectedTransaction, 'originalMessage.IOUTransactionID', 0);
+
+            // API returns transactionID as type String in certain cases (if the transaction was rejected), this
+            // workaround should be removed once this is fixed: https://github.com/Expensify/Expensify/issues/165329
+            return Number(transactionID);
+        });
+
+        const rejectableTransactions = _.filter(actionsForIOUReport, (action) => {
+            if (action.originalMessage.type !== 'create') {
+                return;
             }
-        })}
-    </View>
-);
+            return !rejectedTransactionIDs.includes(action.originalMessage.IOUTransactionID);
+        });
+
+        const rejectableTransactionIDs = _.map(rejectableTransactions, transaction => lodashGet(
+            transaction, 'originalMessage.IOUTransactionID', 0,
+        ));
```

Comment:
@marcaaron Would appreciate your thoughts on this logic. I tried a few things to tidy it up, but nothing led to a huge improvement here. Combining `_.filter` and `_.map` became quite unreadable

I miss RxJava syntax for chaining [the equivalent of] Promises: 
```
this.props.reportActions
    .filter()
    .flatMap()
```

Reply:
Might want to have a poke and check out some `_.chain` examples

https://underscorejs.org/#chain

Reply:
This is perfect, thanks!